### PR TITLE
Basic Templated AI Tagging

### PR DESCRIPTION
**Pull request by JSai23**
Please Review this and let me know if you guys see any problems if this was merged with the main difference being the functionality of the tagReportsAI.js which would make the calls to the AI microservice.

### CODE REVIEW

File: backend/config/secrets.json.example
Comment (codelastnight) [inline backend/config/secrets.json.example:1]: i think its probably important to keep a secrets.json.example somewhere, perhaps we create a new one with new defaults that work for us

File: backend/models/report.js
Comment (codelastnight) [inline backend/models/report.js:1]: i was wondering where other data from the AI like confidence and reasoning would go, would it be an object under tag.value?

**Review by codelastnight — COMMENTED** [review]
looks good 2 me 👍

File: backend/models/report.js
Comment (JSai23) [reply backend/models/report.js:1]: New format can adapt to that I can add confidence values later want to get this up and running before tossing more in but I changed the schema to be dynamic to incorporate more currently its key: {value: ... , rationale: ....}

File: backend/config/secrets.json.example
Comment (JSai23) [reply backend/config/secrets.json.example:1]: I will put back the secrets.json and sample env before we merge

File: src/components/GeneratedTag.tsx
Comment (codelastnight) [inline src/components/GeneratedTag.tsx:1]: going to refactor this out into a custom hook for reusability on the popovers

**Pull request merged**